### PR TITLE
Modernise packaging and CI

### PR DESCRIPTION
**Pull request by ZedThree**
Fixes #29, #32, #39, #49

Also replace most uses of `"some string" + str(number)` with f-strings, because it really bothers me.

`black` and `isort` will now be auto-applied on pushes to `master` -- tests can't fail because of these, they won't interfere with PRs, and they will still always get applied

Use [`ruff`](https://beta.ruff.rs/docs/) instead of `flake8` -- it's way faster and we can use `pyproject.toml`

Use PyPA's new ["Trusted Publisher" thing](https://docs.pypi.org/trusted-publishers/) for publishing -- not tried this before, so hopefully it works

Remove uses of the `builtins` module -- this was backported stuff for Python 2

### CODE REVIEW

File: .github/workflows/formatting.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Formatting (black & isort)
+on:
+  push:
+    branches: master
+    paths:
+      - '**.py'
```
Comment (dschwoerer) [inline .github/workflows/formatting.yml:6]: Can we run this always?

I am not sure whether we have executables without `.py` suffix ...

**Review by dschwoerer — APPROVED** [review]
`isort`  can make tests fails. Have been bitten by that, as it moved some comments, and then mypy was not happy anymore.

As we don't use typing information, this should be fine though :+1: 

I haven't used the publishing workflow either, but lets try.

Thanks, this looks awesome (I also love f-strings) :heart:

**Comment by dschwoerer** [comment]
some required tests are missing. Maybe just remove them all? I only find this useful for auto-merging ...

File: .github/workflows/formatting.yml (hunk 1 above)
Comment (ZedThree) [reply .github/workflows/formatting.yml:6]: Just had a quick check, doesn't look like it in this repo

**Pull request merged**